articles/case_studies/fec-javascript-testing.md

```diff
@@ -0,0 +1,70 @@
+# Testing and refactoring JavaScript with FEC
+
+Until August of this year, the JavaScript for the OpenFEC project had no unit tests. This wasn't because the development team wasn't interested in testing--our backend code was already thoroughly tested. Instead, our JavaScript code just wasn't pleasant to test. Here's an example of the coding style that prevailed not long ago:
+
+```javascript
+$(document).ready(function() {
+  var $widget = $('#widget');
+  $widget.on('click', function() {
+    $.getJSON($widget.data('url')).done(function(response) {
+      $widget.find('.content').text(response.content);
+    }).fail(function(error) {
+      alert('It broke!');
+    });
+  });
+});
+```
+
+Because this is a toy example, it's not too hard to tell what this code does:
+* On document load, bind a click listener to the widget element
+* On click, fetch data from the associated URL
+* If the request succeeds, show the contents in the widget
+* If the request fails, show a warning dialog
+
+But even this simple example is fairly complicated to test. For example, to test that the widget's text is updated on loading data, we would have to stub `$.getJSON` to respond with some known text rather than making a real AJAX request, then trigger the document ready and widget click events. There would be far more code written on test setup and teardown than on the tests themselves, and test-writing would be laborious and error-prone.
```

Comment:
Rather that reading a paragraph about how difficult it would be to test this, I think it would be super helpful to just see how gnarly a test would be via actually showing the test setup required